mailpoet/lib/Automation/Engine/Storage/WorkflowStorage.php

```diff
@@ -10,43 +10,80 @@
 
 class WorkflowStorage {
   /** @var string */
-  private $table;
+  private $workflowTable;
+  /** @var string */
+  private $versionsTable;
 
   /** @var wpdb */
   private $wpdb;
 
   public function __construct() {
     global $wpdb;
-    $this->table = $wpdb->prefix . 'mailpoet_workflows';
+    $this->workflowTable = $wpdb->prefix . 'mailpoet_workflows';
+    $this->versionsTable = $wpdb->prefix . 'mailpoet_workflow_versions';
     $this->wpdb = $wpdb;
   }
 
   public function createWorkflow(Workflow $workflow): int {
-    $result = $this->wpdb->insert($this->table, $workflow->toArray());
+    $result = $this->wpdb->insert($this->workflowTable, $this->workflowHeaderData($workflow));
     if (!$result) {
       throw Exceptions::databaseError($this->wpdb->last_error);
     }
-    return $this->wpdb->insert_id;
+    $id = (int)$this->wpdb->insert_id;
+    $this->insertWorkflowVersion($id, $workflow);
+    return $id;
   }
 
   public function updateWorkflow(Workflow $workflow): void {
-    $result = $this->wpdb->update($this->table, $workflow->toArray(), ['id' => $workflow->getId()]);
+    $result = $this->wpdb->update($this->workflowTable, $this->workflowHeaderData($workflow), ['id' => $workflow->getId()]);
     if ($result === false) {
       throw Exceptions::databaseError($this->wpdb->last_error);
     }
+    $this->insertWorkflowVersion($workflow->getId(), $workflow);
   }
 
   public function getWorkflow(int $id): ?Workflow {
-    $table = esc_sql($this->table);
-    $query = (string)$this->wpdb->prepare("SELECT * FROM $table WHERE id = %d", $id);
+    $workflowTable = esc_sql($this->workflowTable);
+    $versionTable = esc_sql($this->versionsTable);
+    $query = (string)$this->wpdb->prepare("
+SELECT
+      workflow.*,
+      version.id AS version_id,
+      version.steps,
+      version.trigger_keys
+FROM
+      $workflowTable as workflow,
+      $versionTable as version
+WHERE
+      version.workflow_id = workflow.id
+ORDER BY
+      version.id DESC
+LIMIT
+      0,1; ",
+      $id
```

Comment:
Could we indent this similarly to normal PHP code (no left out-dent, sub-indenting by 2 spaces)? We seem to do that [in existing code](https://github.com/mailpoet/mailpoet/blob/c8278dde2a066f1f4602ef32151cc8d4ad22ca90/mailpoet/lib/Newsletter/NewslettersRepository.php#L268) and I guess it may look a tad bit nicer.

This applies also to other similar occurrences in this PR.

Reply:
Let's keep it consistent :+1:
Addressed in [f937c5d](https://github.com/mailpoet/mailpoet/pull/4289/commits/f937c5d5c190c440ec2ffc7ba55bf65ee0296ce8)